Source/WebCore/rendering/RenderWidget.cpp

```diff
@@ -257,12 +257,15 @@ void RenderWidget::paintContents(PaintInfo& paintInfo, const LayoutPoint& paintO
     // to paint itself. That way it will composite properly with z-indexed layers.
     LayoutRect paintRect = paintInfo.rect;
 
-    OptionSet<PaintBehavior> oldBehavior = PaintBehavior::Normal;
-    if (paintInfo.paintBehavior & PaintBehavior::DefaultAsynchronousImageDecode) {
-        if (RefPtr frameView = dynamicDowncast<LocalFrameView>(m_widget)) {
-            oldBehavior = frameView->paintBehavior();
-            frameView->setPaintBehavior(oldBehavior | PaintBehavior::DefaultAsynchronousImageDecode);
-        }
+    static constexpr OptionSet flagsToCopy {
+        PaintBehavior::DefaultAsynchronousImageDecode,
+        PaintBehavior::FlattenCompositingLayers,
+        PaintBehavior::Snapshotting
+    };
+    std::optional<OptionSet<PaintBehavior>> oldBehavior = PaintBehavior::Normal;
```

Comment:
I'm not sure I understand how the `std:optional` is useful here. Would it ever be `std::nullopt` if you set it to `PaintBehavior::Normal` ?

Reply:
Good catch, this was a bug.

I ended up finding a better way to write this entire PR though, so no longer relevant :)